### ISSUE TITLE
Type errors when importing legacy typeorm adapter

### ISSUE THREAD

**Issue by fandy** [nextauthjs/adapters#215]
Getting a `Module '"@next-auth/typeorm-legacy-adapter"' has no exported member 'TypeORMLegacyAdapter'.` when importing the adapter:

```import { TypeORMLegacyAdapter } from "@next-auth/typeorm-legacy-adapter";```

**Comment by balazsorban44** [comment]
will need a bit more info

**Comment by willnode** [comment]
![image](https://user-images.githubusercontent.com/20214420/133492912-6ba2bdd3-e4f1-455d-9e61-e8766ea897f8.png)

This is straight [from the docs](https://next-auth.js.org/adapters/typeorm)

This works, but we'd missing typings?

![image](https://user-images.githubusercontent.com/20214420/133493109-a44464c0-bf85-47e2-8ba7-33e158af1eb7.png)

**Comment by balazsorban44** [comment]
I meant reproduction, package versions, etc.

**Comment by willnode** [comment]
Here's mine

```
dom-new-portal@0.1.0
└─┬ next-auth@3.29.0  
  ├─┬ @next-auth/prisma-legacy-adapter@0.1.2
  │ └── next-auth@3.29.0 deduped
  └─┬ @next-auth/typeorm-legacy-adapter@0.1.4
    └── next-auth@3.29.0 deduped
```

It looks like this function was declared but not in [index.d.ts](https://github.com/nextauthjs/adapters/blob/main/packages/typeorm-legacy/index.d.ts)
https://github.com/nextauthjs/adapters/blob/fae5c50d3accac2c4b380f456ef5ffe96402df6b/packages/typeorm-legacy/src/index.js#L13

**Comment by fandy** [comment]
> I meant reproduction, package versions, etc.

![image](https://user-images.githubusercontent.com/19108762/133522909-ef49bf61-daa9-4b6e-8386-5c349372b4b7.png)

`    "@next-auth/typeorm-legacy-adapter": "^0.1.4",`

**Comment by willnode** [comment]
Ok I definitely mixing up between v3 and upcoming v4. 

@fandy probably you have the same problem with me 😅 installing `next-auth@beta` and `@next-auth/typeorm-legacy-adapter@next` works to me but I have to do some refactor from now.

**Comment by balazsorban44** [comment]
I see. Not sure I want to put any effort in the current adapter, but I'm open for PRs for a fix.

The new version of the Adapter is completely rewritten in TypeScript already: https://github.com/nextauthjs/adapters/tree/next/packages%2Ftypeorm-legacy

**Comment by fandy** [comment]
> Ok I definitely mixing up between v3 and upcoming v4.
> 
> @fandy probably you have the same problem with me 😅 installing `next-auth@beta` and `@next-auth/typeorm-legacy-adapter@next` works to me but I have to do some refactor from now.

Yeah actually just found this solution in the docs (though it's not very visible). @balazsorban44 imo the docs can be more clear by referencing `next-auth/typeorm-legacy-adapter@next` in the v4 docs. Right now it's easy to miss.

**Comment by balazsorban44** [comment]
@fandy there is a quite visible warning in the migration docs that you will have to use `@next`

https://next-auth.js.org/getting-started/upgrade-v4#adapters

**Comment by willnode** [comment]
I remember few weeks ago somehow the documentation has suddenly changed during my first encounter with `next-auth`. Got a lot of problem that I'm mixing stuff already to this today! I though it's only a small fixes 😅 boy I am so wrong.

**Comment by balazsorban44** [comment]
We added a banner to the top when changing the default docs to the new version. 🤔

We added a version selector to every page.

And also a migration guide.

Genuine question, how could we make it more obvious which version of the docs someone is looking at? I've seen many people encountered similar issues, seemingly confused by which version they are trying. 😕

**Comment by fandy** [comment]
> We added a banner to the top when changing the default docs to the new version. 🤔
> 
> We added a version selector to every page.
> 
> And also a migration guide.
> 
> Genuine question, how could we make it more obvious which version of the docs someone is looking at?
> 
> I've seen many people encountered similar issues, seemingly confused by which version they are trying. 😕
![image](https://user-images.githubusercontent.com/19108762/133528255-cffbf047-757e-4473-bd05-00400a4c9e49.png)

On https://next-auth.js.org/adapters/typeorm it recommends installing the version without next, which is the most prominent instruction. I don't see a banner which recommends the `@next` version, but would it even be necessary if we had the install instructions listing `@next`?

**Comment by balazsorban44** [comment]
I see. I kind of assumed that people using v4 will automatically go to the migration docs, where the the info about adding `@next` to adapter installations is very prominent in my opinion. We kind of intended to compress all necessary info into that page, but it looks like people don't really look at it. Most of the recent issues were people not reading it.

I might have been wrong, we might just want to maintain all the beta related changes on all pages for a while. 🤷‍♂️ It creates a bit of extra work, but I don't see a better solution.

**Comment by fandy** [comment]
> I see. I kind of assumed that people using v4 will automatically go to the migration docs, where the the info about adding `@next` to adapter installations is very prominent in my opinion. We kind of intended to compress all necessary info into that page, but it looks like people don't really look at it. Most of the recent issues were people not reading it.
> 
> I might have been wrong, we might just want to maintain all the beta related changes on all pages for a while. 🤷‍♂️ It creates a bit of extra work, but I don't see a better solution.

Yeah that would be a safe way to eliminate the incoming issues, but even just using @next on the main install instructions would be huge. Some users might not be past users of v3, or starting a new project, so they would skimp past migration docs as they are unnecessary.

Ideally we would have all new setup instructions on v4 main docs and all changes on migration docs.

**Comment by balazsorban44** [comment]
I am curious if this banner doesn't show up on some devices, or is not prominent enough?
![](https://i.imgur.com/Adh9lLY.png)

It is supposed to be shown across the WHOLE site.

BTW in the current state, for the core, you would need `@beta`, only the adapters will need `@next`. I didn't bother to add YET another distribution tag for adapters, as it already seems to be hard to follow. 😅

**Comment by fandy** [comment]
I forgot to include the banner in the screenshot, but yes it's showing for me.

Imo that's what makes it confusing - it's telling you that you're viewing the correct instructions for v4, but the installation instructions are outdated. Am I missing something here?

**Comment by willnode** [comment]
> I am curious if this banner doesn't show up on some devices, or is not prominent enough?

To my experience, it's visible enough. But my problem is I'm not aware i'm using v3.x at that time. And to make it worse i'm not even aware these new features are exclusive to `@next`. Yesterday I keep switching between `adapter` and `database` option in config because i'm confused for a whole day why `adapter` option didn't work to me.

I think it all comes with sudden documentation change. Imo URL's should not change just because there's new major release. Weirdly enough, to me this [dead doc URL](https://next-auth.js.org/adapters/typeorm/mysql) is still on top of my address bar suggestions. 

Some large software libraries like Bootstrap or Nodejs already prefix their documentation with software version even if it's the latest. But anyway this are just my thought out of thin air, many software docs don't follow this convention probably because it's easier to do with.

**Comment by willnode** [comment]
@balazsorban44 maybe put some WARN if `database` option is used in v4?  probably a lot of people in future will just copy tutorials old in old versions not realizing it's outdated 😅

**Comment by balazsorban44** [comment]
The docs versioning has been put in place from v4 onward. We didn't have versioning before that. I think our strategy is to not use the version for the most current release for probably SEO reasons. @ndom91 do you have a better explanation?

One mistake I probably made was not including the migration guide in the top banner, which is being fixed in https://github.com/nextauthjs/docs/pull/31
It will look like this:
![](https://i.imgur.com/99GPzbz.png)

We honestly don't do anything fancy with the tools we are given, we use [Docusaurus](https://docusaurus.io/) for docs, and trying our best at [semantic versioning](https://semver.org/). That said, it **is** my first release of this scale and I learn myself along the way.

**Comment by balazsorban44** [comment]
@willnode `database` is gone in v4: https://docs-git-fix-make-beta-more-prominent-nextauthjs.vercel.app/getting-started/upgrade-v4#adapters
